host/sgx/elf.c

```diff
@@ -284,7 +282,7 @@ int elf64_load(const char* path, elf64_t* elf)
     if (fd == -1 || _fstat64(fd, &statbuf) != 0)
         goto done;
 
-    if (!(statbuf.st_mode & _S_IFREG) != 0)
+    if (!(statbuf.st_mode & _S_IFREG))
```

Comment:
Isn't this equivalent to `if ((statbuf.st_mode & _S_IFREG) == 0)`? It's not encouraged to implicitly convert integer to boolean.

Reply:
Yes, it is equivalent. This triggers a warning due to double negation. I could change it to `if ((statbuf.st_mode & _S_IFREG) == 0)`.

For my knowledge, why is it not encouraged to use an int in an if condition?

Reply:
Some compilers with some flags warn about implicit boolean conversion. For example `-Wint-in-bool-context`.